src/ResourceManager/EventHub/Commands.EventHub/Cmdlets/Namespace/SetAzureEventHubNamespaceAuthorizationRules.cs

```diff
@@ -0,0 +1,95 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.EventHub.Models;
+using Microsoft.Azure.Management.EventHub.Models;
+using System.Management.Automation;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.Commands.EventHub.Commands.Namespace
+{
+    /// <summary>
+    /// 'Set-AzureRmEventHubNamespaceAuthorizationRule' Cmdlet updates specified Eventhub Namespace AuthorizationRule
+    /// </summary>
+    [Cmdlet(VerbsCommon.Set, EventHubNamespaceAuthorizationRuleVerb, SupportsShouldProcess = true), OutputType(typeof(SharedAccessAuthorizationRuleAttributes))]
+    public class SetAzureEventHubNamespaceAuthorizationRule : AzureEventHubsCmdletBase
+    {
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = "Resource Group Name.")]
+        [ValidateNotNullOrEmpty]
+         public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 1,
+            HelpMessage = "EventHub Namespace Name.")]
+        [ValidateNotNullOrEmpty]
+        public string NamespaceName { get; set; }
+
+        [Parameter(Mandatory = true,
+            Position = 2,
+            ParameterSetName = SASRuleParameterSetName,
```

Comment:
This is not resolved.  If you don't have more than one parameter set, then remove the ParameterSetName from this attribute.  If you do have multiple parameter sets, then you need to attribute ther parameters for the parameter set they belong in, and you need to set 'DefaultParameterSetName' in the Cmdlet attribute